File: api/models/Expense.js
const mongoose = require('mongoose');

const expenseSchema = new mongoose.Schema({
    amount: {
        type: Number,
        required: [true, 'Please provide the expense amount'],
        min: [0, 'Amount cannot be negative'],
    },
    beneficiary: {
        type: String,
        required: [true, 'Please provide the beneficiary name'],
        trim: true,
    },
    category: {
        type: String,
        required: [true, 'Please provide a category'],
        enum: [
            'Food',
            'Travel',
            'Transfers',
            'Gift',
            'Entertainment',
            'Utilities',
            'Shopping',
            'Healthcare',
            'Education',
            'Other',
        ],
        default: 'Other',
    },
    description: {
        type: String,
        trim: true,
        maxlength: [100, 'Description cannot exceed 100 characters'],
    },
    date: {
        type: Date,
        required: [true, 'Please provide the expense date'],
        default: Date.now,
    },
    user: {
        type: mongoose.Schema.ObjectId,
        ref: 'User',
        required: true,
    },
    createdAt: {
        type: Date,
        default: Date.now,
    },
});

const Expense = mongoose.model('Expense', expenseSchema);
module.exports = Expense;

File: frontend/lib/features/expense/domain/entities/expense.dart
// expense.dart
class Expense {
  final String id;
  final double amount;
  final String beneficiary;
  final String category;
  final String? description;
  final DateTime date;
  final DateTime createdAt;

  Expense({
    required this.id,
    required this.amount,
    required this.beneficiary,
    required this.category,
    this.description,
    required this.date,
    required this.createdAt,
  });

  factory Expense.fromJson(Map<String, dynamic> json) {
    return Expense(
      id: json['_id'],
      amount: json['amount'].toDouble(),
      beneficiary: json['beneficiary'],
      category: json['category'],
      description: json['description'],
      date: DateTime.parse(json['date']),
      createdAt: DateTime.parse(json['createdAt']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'amount': amount,
      'beneficiary': beneficiary,
      'category': category,
      'description': description,
      'date': date.toIso8601String(),
    };
  }
}
